src/test/celoTransaction/celoTransaction.ts

```diff
@@ -0,0 +1,83 @@
+import { autoinject, computedFrom } from "aurelia-framework";
+import { EventAggregator } from "aurelia-event-aggregator";
+import { BigNumber, ethers } from "ethers";
+import { EthereumService } from "services/EthereumService";
+import { DisposableCollection } from "services/DisposableCollection";
+import { TokenService } from "services/TokenService";
+import TransactionsService from "services/TransactionsService";
+import "./celoTransaction.scss";
+
+@autoinject
+export class celoTransaction {
+  private subscriptions = new DisposableCollection();
+
+  constructor(
+    private eventAggregator: EventAggregator,
+    private ethereumService: EthereumService,
+    private tokenService: TokenService,
+    private transactionService: TransactionsService,
+  ) {
+  }
+
+  private userFundingTokenBalance: BigNumber;
+
+  private account: string = null;
+  private balance: BigNumber = null;
+  private checking: boolean = false;
+  private tokenAmountToPay: BigNumber = null;
+  private addressToPay: string;
+
+  private async initialize() {
+    this.account = this.ethereumService.defaultAccountAddress;
+    this.getBalance();
+  }
+
+  private async getBalance() {
+    if (!this.checking) {
+      try {
+        this.checking = true;
+        if (this.account) {
+          const provider = this.ethereumService.readOnlyProvider;
+          this.balance = await provider.getBalance(this.account);
+        } else {
+          this.balance = null;
+        }
+        // tslint:disable-next-line:no-empty
+        // eslint-disable-next-line no-empty
```

Comment:
dont suppose  we could jsut use one of those, no ?

Reply:
also, wouldn't we want to at least log an error then it happens here ? otherwise,the try catch block kinda is pointless, i feel